### PR TITLE
Shuffle helper

### LINKED ISSUE #20: Shuffle helper
```js
shuffle([1, 2, 3, 4, 5]) === [4, 2, 3, 5, 1];
shuffle([1, 2, 3, 4, 5]) === [1, 5, 2, 3, 4];
shuffle([1, 2, 3, 4, 5]) === [3, 1, 4, 5, 2];
```

```hbs
Our lucky winner is: {{get (shuffle participants) "firstObject"}}
Our lucky winner is: {{get (shuffle participants (action "customRandomizer")) "firstObject"}}
```

### PR DESCRIPTION
**Pull request by robbiespeed**
Closes #20

### CODE REVIEW

**Comment by robbiespeed** [comment]
Tests use a fake randomizer passed in to the shuffle helper, because accurately testing randomness is tricky. It's a standard Fisher–Yates shuffle, but I could write a unit test if desired that would test the shuffle with Math.random many times.

File: addon/helpers/shuffle.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  
+  while (count > 1) {
+    rand = Math.floor(randomizer() * (count--));
+    
+    temp = array[count];
+    array[count] = array[rand];
+    array[rand] = temp;
+  }
+  
+  return emberArray(array);
```
Comment (poteto) [inline addon/helpers/shuffle.js]: Could we return a new array? I would prefer to not mutate the `array` argument being passed in.

File: addon/helpers/shuffle.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export default Helper.extend({
+  compute([array, random]) {
+    if (!isArray(array)) {
+      return emberArray([array]);
+    }
+    
+    set(this, 'array', array);
+    return shuffle(array.slice(0), random);
```
Comment (poteto) [inline addon/helpers/shuffle.js]: Once the `shuffle` function is pure, this `slice` is unnecessary

**Comment by poteto** [comment]
Looks good to me, apart from minor comments. Thanks!

**Comment by robbiespeed** [comment]
Updated should be good to go now

**Comment by poteto** [comment]
Sorry about the delay! We just need to add an entry to [`index.js`](https://github.com/DockYard/ember-composable-helpers/blob/master/addon%2Findex.js) and we should be all set. Thanks!

**Comment by poteto** [comment]
@homu r+

**Comment by poteto** [comment]
@homu r-

**Pull request closed without merging**